tests/integration/ambient/testdata/modified-waypoint-template.yaml

```diff
@@ -0,0 +1,1451 @@
+apiVersion: v1
```

Comment:
checking in this entire file seems problematic. It will quickly go out of sync and means that you can only test with the exact same setup as whatever values were used to generate this.

Also, since its not cleaned up, the next time you run the test it will still have this and break the test (I assume - I didn't try it)

Reply:
I thought the test framework automatically cleaned up resources between executions?  I'd be happy to add a cleanup step if that would help.  Can you point me to an example?

As for this file going out of sync, I agree, though I don't see an easy solution.  The templates are buried so deep in the injector config that a minimal patch is not possible (just parsing in bash requires piping yq output to yq, as we have yaml nested in a string in another yaml).  Long term, I believe that the correct solution is to extract templates into either their own CRDs, or their own config maps.  That will allow patching a single field in the config, which is really what we want to do here.  Do you have any recommendations how we might achieve similar results in the short term?

Reply:
For local development you typically run with `--istio.test.nocleanup` so you don't have to spend 1 min deploying Istio between runs. Also for vendors and custom installs, you would do nocleanup and nodeploy. So messing with the Istio install is problematic.

Does the actual contents of the cm have to change? Or can we just do something like annotate the configmap? I guess that would trigger a NOP Apply call which we don't test?

Reply:
We need to write some change that can be detectable in the output deployment.  I'm open to options.  I could cache the value of the CM before modifying, and restore the old version at the end of the test.  But that won't run if the test fails.  Go test frameworks leave much to be desired in terms of setup and cleanup capabilities.

Reply:
`t.cleanup` - we actually do this for mesh config, I forgot: https://github.com/istio/istio/blob/dd1f9f5a5373b9117fd55e59a29e763429b58fbf/pkg/test/framework/components/istio/configmap.go#L98

It doesn't run on panic I think, but a proper test doesn't panic